operator/src/main/java/org/keycloak/operator/v2alpha1/KeycloakController.java

```diff
@@ -41,11 +41,13 @@
 import javax.inject.Inject;
 import java.util.List;
 import java.util.Optional;
+import java.util.concurrent.TimeUnit;
 
 import static io.javaoperatorsdk.operator.api.reconciler.Constants.NO_FINALIZER;
 import static io.javaoperatorsdk.operator.api.reconciler.Constants.WATCH_CURRENT_NAMESPACE;
 
-@ControllerConfiguration(namespaces = WATCH_CURRENT_NAMESPACE, finalizerName = NO_FINALIZER)
+// TODO: remove "generationAwareEventProcessing = false" when the race condition is fixed
+@ControllerConfiguration(namespaces = WATCH_CURRENT_NAMESPACE, finalizerName = NO_FINALIZER, generationAwareEventProcessing = false)
```

Comment:
This is the key change, disabling `generationAwareEventProcessing` workaround the race condition we found.

Reply:
Can you please create a follow-up task with all the linked issues that need to be resolved to unblock it?

Reply:
Follow up issue #10658